test/e2e/destroypod_test.go

```diff
@@ -178,6 +178,17 @@ func TestDestroyPodTimely(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Failed to create a service: %v", err)
 	}
+	routeURL := objects.Route.Status.URL.URL()
+
+	if _, err = pkgTest.WaitForEndpointState(
```

Comment:
Why WaitFor and not Check?

Reply:
I don't see a function for check endpoint state instead of wait for. It's also what the other tests are using so I wanted to keep it consistent.

Reply:
We *should* have a Check dual for every WaitFor function, but you are right that in this case we don't yet.  In this case, we mask "Ready doesn't mean Ready" errors by polling when we don't get what we expect.

Reply:
I see. I agree this is a problem. I'm seeing this in many other tests (creating a ready service and then waiting for endpoint to be ready instead of checking). Are we looking to use check for other tests as well?

Reply:
See https://github.com/knative/serving/issues/5573 for what's blocking us from using a check (almost) everywhere.

Reply:
@markusthoemmes I see. Thanks for the context! Does that mean we still cannot use a check here until that's resolved?